csclient/csclient.go

```diff
@@ -234,21 +234,26 @@ func (c *Client) ListResources(id *charm.URL) ([]params.Resource, error) {
 
 // Progress lets an upload notify a caller about the progress of the upload.
 type Progress interface {
-	// Start is called with the size of the upload and its upload id
-	// when the upload starts. The uploadId will be empty when
-	// the upload is small and multipart upload is not being used.
+	// Start is called with the its upload id and when the upload
+	// starts. The upload id will be empty when the upload is small
+	// and multipart upload is not being used.
 	Start(uploadId string, expires time.Time)
 
 	// Transferred is called periodically to notify the caller that
 	// the given number of bytes have been uploaded.
+	// Note that the number may not always increase - for
```

Comment:
It looks like it might well decrease, which is a bit different from just not increasing. You should probably also mention that explicitly.